paperqa/readers.py

```diff
@@ -308,65 +309,74 @@ def read_doc(
     overlap: int = 100,
     page_size_limit: int | None = None,
 ) -> list[Text] | ParsedText | tuple[list[Text], ParsedMetadata]:
-    """Parse a document and split into chunks.
+    """Parse a document and split into chunks without blocking the event loop.
 
-    Optionally can include just the parsing as well as metadata about the parsing/chunking
-
-    Args:
-        path: local document path
-        doc: object with document metadata
-        parsed_text_only: return parsed text without chunking
-        include_metadata: return a tuple
-        chunk_chars: size of chunks
-        overlap: size of overlap between chunks
-        page_size_limit: optional limit on the number of characters per page
+    Offloads blocking I/O and CPU operations to a thread pool.
     """
     str_path = str(path)
-    parsed_text = None
 
     # start with parsing -- users may want to store this separately
     if str_path.endswith(".pdf"):
-        parsed_text = parse_pdf_to_pages(path, page_size_limit=page_size_limit)
+        parsed_text = await asyncio.to_thread(
```

Comment:
Can you add `TODO` comments next to `parse_pdf_to_pages` and `parse_text` to make them `async`